### PR TITLE
add common usage examples

### PR DESCRIPTION
**Pull request by kbrock**
It was not clear before how these commands fit together.

Here is how I understood how our process worked.
Don't think I dumbed it down too much but please let me know

### CODE REVIEW

File: README.md
Comment (Fryguy) [inline README.md:39]:
```suggestion
You can either downloading the latest build image or build a new one locally:
```

File: README.md
Comment (Fryguy) [inline README.md:37]: This seems unclear...instead can we use

```suggestion
## Obtain container image for building RPMs
```

File: README.md
Comment (Fryguy) [inline README.md:53]: This has a -v but doesn't pass an actual volume...was it missed?  (Same with the next command)

File: README.md
Comment (Fryguy) [inline README.md:66]:
```suggestion
The [options.yml](blob/master/config/options.yml) points to the appropriate repos and tag/branches.
```

File: README.md
Comment (Fryguy) [inline README.md:123]: What is this referencing?

File: README.md
Comment (Fryguy) [inline README.md:120]: I don't understand what this means - I've never needed the volumes in docker preferences as far as I can remember.

File: README.md
Comment (Fryguy) [inline README.md:134]: There's an extra -v after the run.

File: README.md
Comment (Fryguy) [inline README.md:146]: it's generally a lot easier if you're running a daemonized container to give it a name using `--name`.  Then later you just reference the container by name directly, and then no need for env vars or anything.

File: README.md
Comment (Fryguy) [inline README.md:182]: This is generally the easiest way if you're using a daemonized container...I think we should just remove the "Pushing the build artifacts to a mounted volume" section, personally, to simplify (too many choices => paradox of choice)

**Review by Fryguy — APPROVED** [review]
Overall, LGTM...would like @bdunne to also review.

File: README.md
Comment (bdunne) [inline README.md:31]:
```suggestion
* `/var/www/miq/vmdb` contains application files.
```

File: README.md
Comment (bdunne) [inline README.md:46]: Should we dockerignore Gemfile.lock?

File: README.md
Comment (bdunne) [reply README.md:120]: Is this a Docker on Mac thing?

File: README.md
Comment (kbrock) [reply README.md:123]: if you want to add a mounted volume using `-v`, then you will need to add the volume to your file share and restart docker.

File: README.md
Comment (kbrock) [reply README.md:120]: yea mac thing

File: README.md
Comment (kbrock) [reply README.md:182]: to be honest, I tend to go in, build it, an while I'm in there just copy to /root/OPTIONS

I'd prefer just a `--copy-rpms $TARGET` option using build and be done with it.

Comment (bdunne) [reply README.md:182]: > to be honest, I tend to go in, build it, an while I'm in there just copy to /root/OPTIONS
> 
> I'd prefer just a `--copy-rpms $TARGET` option using build and be done with it.

Wouldn't it be easier to use a second volume mount rather than daemonize, exec into it, then copy to a volume mount?

File: README.md
Comment (Fryguy) [reply README.md:46]: Maybe?  In production builds we _want_ to bring a Gemfile.lock during rpm build because we want to ensure we aren't arbitrarily updating gems.  We usually have a Gemfile.lock.release file, but I don't know how @simaishi made that be the "real" lock file.

File: README.md
Comment (kbrock) [reply README.md:123]: thnx - verified - removed

File: README.md
Comment (kbrock) [reply README.md:182]: 1. removing push
2. I find it easier to mount all these directories but I will try again and see how it goes. I've already been convinced of a number of changes that are making this easier

File: README.md
Comment (kbrock) [reply README.md:120]: thanks for solving this. With this knowledge, the `-v` option is much better

**Comment by kbrock** [comment]
@bdunne Could you double check my usage for building jansa above (or we could update to k or l - up to you)

How do you build an old release? Do you use `rpm_build:latest` or `latest-$release`, or another? And which `--git-ref` do you use?  Maybe `jansa1`?

```sh
docker run --rm manageiq/rpm_build:latest-jansa build --git-ref jansa1 --update-rpm-repo
```

**Comment by bdunne** [comment]
Yes, for older branches we use `latest-jansa`.  I don't think we need the `--git-ref` for the older branches since it's set [here](https://github.com/ManageIQ/manageiq-rpm_build/blob/lasker/config/options.yml#L4).

File: README.md
Comment (kbrock) [reply README.md:153]: thanks for the question. Misunderstandings of previous errors had me going in a tangent.

File: README.md
Comment (Fryguy) [reply README.md]: Oh I misunderstood this - I was thinking the Gemfile.lock was the _application's_ Gemfile.lock as opposed to the one for manageiq-rpm_build itself.  Yeah, for the latter, we can probably .dockerignore that.

**Comment by kbrock** [comment]
ok, updated the statement around `--git-ref` not being for tagged builds.

I'm good to go with this one if others are as well.

File: README.md
Comment (Fryguy) [inline README.md:110]:
```suggestion
- `/root/BUILD/rpms/x86_64/manageiq-*.rpm`
```

File: README.md
Comment (Fryguy) [inline README.md:125]: We probably should mark this as being Docker on Mac specific...this steps doesn't make sense for normal Docker.  I'm ok with a follow-up PR though.

**Comment by Fryguy** [comment]
LGTM - one typo.  I'll let @bdunne merge.

File: README.md
Comment (kbrock) [reply README.md:125]: thanks - I'll dump this section. now that you found out that you do not need to put every directory into there, this makes the use case much easier

File: README.md
Comment (bdunne) [inline README.md:56]:
```suggestion
For older builds, the `git-ref` is already stored in `config/options.yml` and not necessary.
```

File: README.md
Comment (bdunne) [inline README.md:92]:
```suggestion
Alternatively, a static directory may be used like `/opt/OPTIONS/options.yml`.
```

File: README.md
Comment (bdunne) [inline README.md:112]:
```suggestion
Depending upon your use case, there are a few ways to obtain these artifacts:
```

File: README.md
Comment (bdunne) [inline README.md:120]:
```suggestion
Since the files are not uploaded, they need to be retrieved from the repo before the container exits.
```

Comment (bdunne) [reply README.md:120]: IIRC, all of the files still exist on the mounted directory after the container exits, correct?

Comment (kbrock) [reply README.md:120]: once the primary process exists, you can not access the container anymore.

**Comment by kbrock** [comment]
ok:

- made all doc changes.
- introduced dockerignore with Gemfile.lock
- outstanding: how can you get files out of a container that has exited the primary process?

Think we are done and ready to go. Pretty sure point 3 above is not possible

**Comment by Fryguy** [comment]
> outstanding: how can you get files out of a container that has exited the primary process?

1. don't use `--rm` (which causes the stopped container to be removed)
2. Use `docker cp` to copy the files
3. `docker rm` to remove the stopped container

File: README.md
Comment (kbrock) [reply README.md]: I think I'm wrong here. I'll revisit and fix.

Since I am wrong, this greatly simplifies things. (thanks to jason't comments down below)

**Comment by kbrock** [comment]
ok, pushed again, removing a bunch of complexity I had added

mostly surrounding copying assets out of a container. `--rm` was the culprit for many of my difficulties

**Review by Fryguy — APPROVED** [review]
LGTM - @bdunne is this good?

**Review by bdunne — APPROVED** [review]

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 492becd490fa0ebe00757dfa4b51fdf2e3ae5171.

```text
commit 492becd490fa0ebe00757dfa4b51fdf2e3ae5171
Author: Brandon Dunne <bdunne@redhat.com>
Date:   Wed Jun 16 15:41:09 2021 -0400

    Merge pull request #152 from kbrock/document_build
    
    add common usage examples
    
    (cherry picked from commit f9908ca9eb9827729963a8a00c9d2e619066f29e)
```

**Pull request merged**
